### PR TITLE
Added Controllers for User, Order, ShoppingCart and Authentication

### PR DESCRIPTION
**Pull request by vicire**

### CODE REVIEW

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/cinema/controller/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import cinema.model.dto.request.UserRequestDto;
+import cinema.security.AuthenticationService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/controller/AuthenticationController.java]: what is the point of this annotation here?

File: src/main/java/cinema/model/dto/response/UserResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package cinema.model.dto.response;
+
+public class UserResponseDto {
+    private Long userId;
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/model/dto/response/UserResponseDto.java]:
```suggestion
    private Long id;
```

File: src/main/java/cinema/service/mapper/ResponseDtoMapper.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package cinema.service.mapper;
+
+public interface ResponseDtoMapper<T, V> {
+    T toDto(V v);
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/service/mapper/ResponseDtoMapper.java]:
```suggestion
    T toDto(V entity);
```

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/cinema/controller/AuthenticationController.java (hunk 1 above)
Comment (vicire) [reply src/main/java/cinema/controller/AuthenticationController.java]: Fixed

File: src/main/java/cinema/model/dto/response/UserResponseDto.java (hunk 2 above)
Comment (vicire) [reply src/main/java/cinema/model/dto/response/UserResponseDto.java]: Fixed

File: src/main/java/cinema/service/mapper/ResponseDtoMapper.java (hunk 3 above)
Comment (vicire) [reply src/main/java/cinema/service/mapper/ResponseDtoMapper.java]: Fixed

**Pull request merged**